### PR TITLE
Konstantin/appeals 57279

### PR DESCRIPTION
**Pull request by KonstantinShevtsov**
Resolves [APPEALS-57279](https://jira.devops.va.gov/browse/APPEALS-57279)

# Description
That PR adding error handling for all cases where in RequestIssuesUpdateEvent class searching request_issue by reference_id. It also optimizes and fixes some of the methods in RequestIssuesUpdateEvent class, adding some logic into app/services/events/decision_review_updated.rb adds and updates rspec tests for RequestIssuesUpdateEvent class.

## Acceptance Criteria
- [x] Code compiles correctly

# Best practices
## Code Documentation Updates
- [x] Add or update code comments at the top of the class, module, and/or component.

### Test Coverage
Did you include any test coverage for your code? Check below:
- [x] RSpec

### Code Climate
Your code does not add any new code climate offenses? If so why?
- [x] No new code climate issues added

### Error Handling
- [x] Are errors being caught and handled gracefully?
- [x] Are appropriate error messages being displayed to users?
- [x] Are critical errors being reported to an error tracking system (e.g., Sentry, ELK)?
- [x] Are unhandled exceptions being caught at the application level ?

### CODE REVIEW

File: app/services/events/decision_review_updated.rb [hunk 1]
```diff
@@ -2,11 +2,20 @@
...
+
+        review = EndProductEstablishment.find_by(
+          reference_id: parser.end_product_establishments_reference_id
+        )&.source
+
+        RequestIssuesUpdateEvent.new(user: user, review: review, parser: parser).perform!
+
         event = find_or_create_event(consumer_event_id)
 
         # Events::DecisionReviewUpdated::UpdateInformalConference.process!(event: event, parser: parser)
```
Comment (nkutub) [inline app/services/events/decision_review_updated.rb:21]: we should uncoment these two lines to complete the entire process

Comment (KonstantinShevtsov) [reply app/services/events/decision_review_updated.rb]: > we should uncoment these two lines to complete the entire process

done.

File: app/models/request_issues_update_event.rb [hunk 2]
```diff
@@ -133,122 +149,126 @@ def from_intake_data(data, decision_review: nil)
...
-    return if edited_issues.empty?
-
-    edited_issue_data.each do |edited_issue|
-      request_issue = RequestIssue.find(edited_issue[:reference_id].to_s)
-      edit_contention_text(edited_issue, request_issue)
-      edit_decision_date(edited_issue, request_issue)
-    end
-  end
-
   def process_eligible_to_ineligible_issues!
```
Comment (nkutub) [inline app/models/request_issues_update_event.rb:215]: Inconsistent variable naming:
The method uses eligible_to_ineligible_issue, but then tries to access ineligible_to_ineligible_issue.closed_at. This is likely a bug and should be eligible_to_ineligible_issue.closed_at.
Incorrect attribute access:
The code uses eligible_to_ineligible_issue.ineligible_reason, but eligible_to_ineligible_issue is a hash, not an object. It should be eligible_to_ineligible_issue[:ineligible_reason].
3. Logic inconsistency:
The method skips updating if the ineligible_reason is not nil, which is the opposite of what we'd expect for an eligible-to-ineligible transition.
Missing error handling:
There's no handling for the case where ineligible_reason or closed_at might be missing from the hash.

Comment (KonstantinShevtsov) [reply app/models/request_issues_update_event.rb:216]: Fixed

File: spec/models/request_issues_update_event_spec.rb [hunk 3]
```diff
@@ -2,24 +2,21 @@
 
 RSpec.describe RequestIssuesUpdateEvent, type: :model do
```
[the remark this answers is not in the document]
Comment (KonstantinShevtsov) [reply spec/models/request_issues_update_event_spec.rb:3]: fixed, 13 new tests added

**Review by nkutub — APPROVED** [review]

**Pull request merged**